src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -741,56 +743,99 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
+         if(!isPlainInterval(*a))
+            return false;
+
+         auto it = std::find_if(
+            rightIntervals.begin(),
+            rightIntervals.end(),
+            [&](const auto& b)
+            {
+               return a->Start() == b->Start() &&
+                  a->End() == b->End() &&
+                  isPlainInterval(*b);
+            });
+         if(it == rightIntervals.end())
+            return false;
+      }
+      return true;
+   };
+
+   if(RealtimeEffectList::Get(*left).GetStatesCount() != 0 ||
+      RealtimeEffectList::Get(*right).GetStatesCount() != 0 ||
+      !checkAligned(*left, *right))
+   {
+      const auto answer = BasicUI::ShowMessageBox(
+         XO(
+"The tracks you are attempting to merge to stereo contain clips at\n"
+"different positions, or otherwise mismatching clips. Merging them\n"
+"will render the tracks.\n\n" 
+"This causes any realtime effects to be applied to the waveform and\n"
+"hidden data to get removed. Additionally, clips may get padded with\n"
+"silence to match them up.\n\n"
+"Do you wish to continue?"
+         ),
          BasicUI::MessageBoxOptions{}
-            .Caption(XO("Error"))
-            .IconStyle(BasicUI::Icon::Error));
-      return;
+            .ButtonStyle(BasicUI::Button::YesNo)
+            .Caption(XO("Combine mono to stereo")));
+      if(answer != BasicUI::MessageBoxResult::Yes)
+         return;
    }
 
-   bool bBothMinimizedp =
-      ((ChannelView::Get(*pTrack->GetChannel(0)).GetMinimized()) &&
-       (ChannelView::Get(*partner->GetChannel(0)).GetMinimized()));
-
-   tracks.MakeMultiChannelTrack( *pTrack, 2, false );
-
-   pTrack->SetPan( 0.0f );
-
-   // Set NEW track heights and minimized state
-   auto
-      &view = WaveChannelView::Get(*pTrack),
-      &partnerView = WaveChannelView::Get(*partner);
-   view.SetMinimized(false);
-   partnerView.SetMinimized(false);
-   int AverageHeight = (view.GetHeight() + partnerView.GetHeight()) / 2;
-   view.SetExpandedHeight(AverageHeight);
-   partnerView.SetExpandedHeight(AverageHeight);
-   view.SetMinimized(bBothMinimizedp);
-   partnerView.SetMinimized(bBothMinimizedp);
-
+   const auto viewMinimized =
+      ChannelView::Get(*left->GetChannel(0)).GetMinimized() &&
+      ChannelView::Get(*right->GetChannel(0)).GetMinimized();
+   const auto averageViewHeight =
+      (WaveChannelView::Get(*left).GetHeight() +
+      WaveChannelView::Get(*right).GetHeight()) / 2;
+
+   left->SetPan(-1.0f);
+   right->SetPan(1.0f);
+   auto mix = MixAndRender(
+      TrackIterRange {
+         tracks.Any<const WaveTrack>().find(left),
+         ++tracks.Any<const WaveTrack>().find(right)
+      },
+      Mixer::WarpOptions{ tracks.GetOwner() },
+      (*first)->GetName(),
+      &WaveTrackFactory::Get(*project),
+      //use highest sample rate
+      std::max(left->GetRate(), right->GetRate()),
+      //use widest sample format
+      std::max(left->GetSampleFormat(), right->GetSampleFormat()),
+      0.0, 0.0);
```

Comment:
You rely on `MixAndRender` using instead the minimum start and maximum end times of the range, when it is given an empty interval of times.

I had forgot that behavior and stepped in the debugger to convince myself this call works.

The header file for `MixAndRender` should comment this behavior.